index.js

```diff
@@ -228,10 +235,30 @@ module.exports = function (schema, options) {
         });
     }
 
-    schema.methods.delete = function (deletedBy, cb) {
-        if (typeof deletedBy === 'function') {
-          cb = deletedBy;
-          deletedBy = null;
+    schema.methods.delete = function (params, cb) {
+        let deletedBy = null;
+        let deletedId = null;
+
+        if (typeof params === 'function') {
+          cb = params;
+          params = {};
+        }
+
+        // If options is a string or ObjectId, it's the deletedBy value
+        if (typeof params === 'string' || typeof params === 'object' && !params.deletedId ||  (params && params._bsontype === 'ObjectID')) {
```

Comment:
❓ I don't think you need to check if params is an object here as you do that later on
```suggestion
        if (
            typeof params === 'string' || 
            params && params._bsontype === 'ObjectID'
        ) {
```